### PR TITLE
chore(workspace)!: use Application, Service, and Copilot terms

### PR DESCRIPTION
**Pull request by efekarakus**
BREAKING CHANGE: The following user-visible changes are introduced:
1. The `ecs-project` directory is renamed to `copilot`.
2. The `.ecs-workspace` summary file is renamed to `.workspace`.
3. Errors include the term "service" and "application".

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: internal/pkg/workspace/workspace.go [hunk 1]
```diff
@@ -40,18 +39,17 @@ const (
...
-	projectDir string
+	copilotDir string
 	fsUtils    *afero.Afero
 }
 
-// New returns a workspace, used for reading and writing to
-// user's local workspace.
+// New returns a workspace, used for reading and writing to user's local workspace.
 func New() (*Workspace, error) {
 	appFs := afero.NewOsFs()
```
Comment (iamhopaul123) [inline internal/pkg/workspace/workspace.go:54]: Shall we rename this variable `appFs`?

File: internal/pkg/workspace/workspace.go [hunk 2]
```diff
@@ -191,39 +187,39 @@ func (ws *Workspace) WritePipelineManifest(marshaler encoding.BinaryMarshaler) (
 	return ws.write(data, pipelineFileName)
 }
 
-// DeletePipelineManifest removes the from the project directory.
```
Comment (iamhopaul123) [inline internal/pkg/workspace/workspace.go:194]: lol "remove the from the ..."

File: internal/pkg/workspace/workspace_test.go [hunk 3]
```diff
@@ -149,50 +149,50 @@ func TestWorkspace_Summary(t *testing.T) {
...
+				appFS.MkdirAll("test/copilot", 0755)
+				afero.WriteFile(appFS, "test/copilot/.workspace", []byte(fmt.Sprintf("---\napplication: %s", "DavidsApp")), 0644)
 			},
 		},
 		"existing workspace and workspace summary in different directory": {
 			workingDir:     "test/app/",
-			projectName:    "DavidsProject",
+			appName:        "DavidsApp",
 			expectNoWrites: true,
 			mockFileSystem: func(appFS afero.Fs) {
```
Comment (iamhopaul123) [inline internal/pkg/workspace/workspace_test.go:171]: Same here for `appFs`.

File: internal/pkg/workspace/workspace_test.go [hunk 4]
```diff
@@ -508,7 +508,7 @@ func TestWorkspace_DeletePipeline(t *testing.T) {
 			// There should be no pipeline manifest in the project
```
Comment (iamhopaul123) [inline internal/pkg/workspace/workspace_test.go:508]: project!

File: internal/pkg/workspace/workspace_test.go [hunk 5]
```diff
@@ -524,34 +524,34 @@ func TestWorkspace_DeletePipeline(t *testing.T) {
 func TestWorkspace_ReadAddonsDir(t *testing.T) {
 	testCases := map[string]struct {
 		appName        string
```
Comment (iamhopaul123) [inline internal/pkg/workspace/workspace_test.go:526]: maybe `svcName`?

File: internal/pkg/workspace/workspace_test.go [hunk 6]
```diff
@@ -418,10 +418,10 @@ func TestWorkspace_DeleteApp(t *testing.T) {
 		"deletes an non-existing app": {
```
Comment (iamhopaul123) [inline internal/pkg/workspace/workspace_test.go:418]: "non-existing svc"

File: internal/pkg/workspace/workspace_test.go [hunk 7]
```diff
@@ -395,21 +395,21 @@ func TestWorkspace_Write(t *testing.T) {
...
-func TestWorkspace_DeleteApp(t *testing.T) {
+func TestWorkspace_DeleteService(t *testing.T) {
 	testCases := map[string]struct {
 		name string
 
-		projectDir string
+		copilotDir string
 		fs         func() afero.Fs
 	}{
 		"deletes existing app": {
```
Comment (iamhopaul123) [inline internal/pkg/workspace/workspace_test.go:405]: "existing svc"

File: internal/pkg/workspace/workspace_test.go [hunk 8]
```diff
@@ -355,15 +355,15 @@ func TestWorkspace_Write(t *testing.T) {
 	}{
 		"create file under nested directories": {
 			elems:      []string{"webhook", "addons", "policy.yml"},
-			wantedPath: "/ecs-project/webhook/addons/policy.yml",
+			wantedPath: "/copilot/webhook/addons/policy.yml",
 		},
 		"create file under project directory": {
```
Comment (iamhopaul123) [inline internal/pkg/workspace/workspace_test.go:360]: "application directory"

File: internal/pkg/workspace/workspace_test.go [hunk 9]
```diff
@@ -613,7 +613,7 @@ func TestWorkspace_ReadPipelineManifest(t *testing.T) {
 			// GIVEN
 			fs := tc.fs()
 			ws := &Workspace{
-				projectDir: projectDir,
+				copilotDir: projectDir,
```
Comment (iamhopaul123) [inline internal/pkg/workspace/workspace_test.go:616]: also `projectDir` variable name

Comment (efekarakus) [reply internal/pkg/workspace/workspace_test.go:616]: OK NO MORE PROJECT IN THE TESTS! 💣💥

File: internal/pkg/workspace/workspace_test.go (hunk 8 above)
Comment (efekarakus) [reply internal/pkg/workspace/workspace_test.go]: Done!

File: internal/pkg/workspace/workspace_test.go (hunk 7 above)
Comment (efekarakus) [reply internal/pkg/workspace/workspace_test.go]: Done!

**Pull request merged**